main.nf

```diff
@@ -1242,39 +1414,38 @@ process busco {
 ch_busco_multiqc = ch_busco_multiqc.map{it[2]}
 ch_busco_to_summary = ch_busco_to_summary.map{it[2]}
 
-// group by assembler and sample for plotting
+// group by assembler and sample name for plotting
 ch_busco_plot = ch_busco_plot.groupTuple(by: [0,1])
 
 process busco_plot {
-    tag "$assembler-$sample"
+    tag "$assembler-$name"
     publishDir "${params.outdir}/GenomeBinning/QC/BUSCO/", mode: params.publish_dir_mode
 
     input:
-    set val(assembler), val(sample), file(summaries) from ch_busco_plot
+    set val(assembler), val(name), file(summaries) from ch_busco_plot
 
     output:
-    file("${assembler}-${sample}-busco_figure.png")
-    file("${assembler}-${sample}-busco_figure.R")
-    file("${assembler}-${sample}-busco_summary.txt")
+    file("${assembler}-${name}-busco_figure.png")
+    file("${assembler}-${name}-busco_figure.R")
+    file("${assembler}-${name}-busco_summary.txt")
 
     script:
-    def name = "${assembler}-${sample}"
     """
     # replace dots in bin names within summary file names by underscores
     # currently (BUSCO v4.1.3) generate_plot.py does not allow further dots
     for sum in ${summaries}; do
-        [[ \${sum} =~ short_summary.(.*).${name}.(.*).txt ]];
+        [[ \${sum} =~ short_summary.(.*).${assembler}-${name}.(.*).txt ]];
         db_name=\${BASH_REMATCH[1]}
-        bin="${name}.\${BASH_REMATCH[2]}"
+        bin="${assembler}-${name}.\${BASH_REMATCH[2]}"
         bin_new="\${bin//./_}"
         mv \${sum} short_summary.\${db_name}.\${bin_new}.txt
     done
     generate_plot.py --working_directory .
 
-    mv busco_figure.png ${assembler}-${sample}-busco_figure.png
-    mv busco_figure.R ${assembler}-${sample}-busco_figure.R
+    mv busco_figure.png ${assembler}-${name}-busco_figure.png
+    mv busco_figure.R ${assembler}-${name}-busco_figure.R
 
-    summary_busco.py short_summary.*.txt > ${assembler}-${sample}-busco_summary.txt
+    summary_busco.py short_summary.*.txt > ${assembler}-${name}-busco_summary.txt
     """
```

Comment:
Until now it seems that there were everywhere quotes around variables, but here not, is that intended?

Reply:
true, would be nicer to add

Reply:
addressed in #154